library/ios_bgp.py

```diff
@@ -0,0 +1,342 @@
+          - Specify networks to announce via BGP
+        suboptions:
+          network:
+            description:
+              - Network ID to announce via BGP
+            required: True
+          mask:
+            description:
+              - Subnet Mask for the Network to announce
+          route_map:
+            description:
+              - Route-map to modify the attributes
+          state:
+            description:
+              - Specifies the state of network
+            default: present
+            choices:
+              - present
+              - absent
+      af_neighbors:
+        description:
+          - Specifies Address Family neighbor related configurations
+          - This option extends the top-level parameter - neighbors 
+        suboptions:
+          activate:
+            description:
+              - Enable the Address Family for this Neighbor
+            type: bool
+          remove_private_as:
+            description:
+              - Remove private AS number from outbound updates
+            type: bool
+          route_map:
+            description:
+              - Specify the route map to apply to this neighbor
+          route_map_dir:
+            description:
+              - Specify the direction of route map to be applied
+            choices:
+              - in
+              - out
+            default: in
+          route_server_client:
+            description:
+              - Configure a neighbor as Route Server client
+            type: bool
+          weight:
+            description:
+              - Specify default weight for routes from this neighbor
+            type: int
+          next_hop_self:
+            description:
+              - Enable/Disable the next hop calculation for this neighbor
+            type: bool
+          next_hop_unchanged:
+            description:
+              - Enable/Disable propagation of next hop unchanged for iBGP paths to this neighbor
+            type: bool
+      auto_summary:
+        description:
+          - Enable automatic network number summarization
+        type: bool
+      synchronization:
+        description:
+          - Enable IGP synchronization
+        type: bool
+      state:
+        description:
+          - Specifies the state of address family
+        default: present
+        choices:
+          - present
+          - absent
+  state:
+    description:
+      - Specifies the state of the BGP process configured on the device
+    default: present
+    choices:
+      - present
+      - absent
+      - replace
+extends_documentation_fragment: ios
+"""
+
+EXAMPLES = """
+- name: configure global bgp as 65000
+  ios_bgp:
+    bgp_as: 65000
+    router_id: 1.1.1.1
+    log_neighbor_changes: True
+    neighbors:
+      - neighbor: 182.168.10.1
+        remote_as: 500
+      - neighbor: 192.168.20.1
+        remote_as: 500
+    networks:
+      - network: 10.0.0.0/8
+      - network: 11.0.0.0/8
+    state: present
+
```

Comment:
Question here - is it obvious to an IOS network person how the IOS CLI commands convert easily into a task like this?  I'm thinking about more complex tasks that exercise a lot of those sub options (or sub sub options??) and I think the IOS equivalent cli commands are a bit flatter in structure?  I'm not sure, just going on very old memory here...